src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -101,6 +101,18 @@ override protected ImmutableArray<LocalSymbol> BuildLocals()
             return builder.ToImmutableAndFree();
         }
 
+        protected override ImmutableArray<LocalFunctionSymbol> BuildLocalFunctions()
+        {
+            var builder = ArrayBuilder<LocalFunctionSymbol>.GetInstance();
+
+            foreach (var section in _switchSyntax.Sections)
+            {
+                builder.AddRange(BuildLocalFunctions(section.Statements));
```

Comment:
Actually, I think we want the local functions defined in one switch section to be scoped to that switch section. That's because there would be a definite assignment problem otherwise if the local function were referenced from a switch section where the things it captures had not been definitely defined (the frame does not exist).

I know this is a big change. We probably need to do it for pattern matching too. Perhaps leave a TODO about the issue, but there is no way for you to currently generate correct code in the face of functions used across switch sections.